src/analysis/processing/qgsalgorithmexporttopostgresql.cpp

```diff
@@ -0,0 +1,202 @@
+
+  const QString connectionName = parameterAsConnectionName( parameters, QStringLiteral( "DATABASE" ), context );
+
+  try
+  {
+    QgsProviderMetadata *md = QgsProviderRegistry::instance()->providerMetadata( QStringLiteral( "postgres" ) );
+    mConn.reset( static_cast<QgsAbstractDatabaseProviderConnection *>( md->createConnection( connectionName ) ) );
+  }
+  catch ( QgsProviderConnectionException & )
+  {
+    QgsProcessingException( QObject::tr( "Could not retrieve connection details for %1" ).arg( connectionName ) );
+  }
+
+  mSchema = parameterAsSchema( parameters, QStringLiteral( "SCHEMA" ), context );
+  mPrimaryKeyField = parameterAsString( parameters, QStringLiteral( "PRIMARY_KEY" ), context );
+  mEncoding = parameterAsString( parameters, QStringLiteral( "ENCODING" ), context );
+  mOverwrite = parameterAsBoolean( parameters, QStringLiteral( "OVERWRITE" ), context );
+
+  mTable = parameterAsDatabaseTableName( parameters, QStringLiteral( "TABLENAME" ), context );
+  if ( !mTable.isEmpty() )
+  {
+    mTable = mTable.trimmed();
+  }
+  else
+  {
+    mTable = mSource->sourceName();
+    mTable = mTable.replace( QStringLiteral( "." ), QStringLiteral( "_" ) );
+  }
+  mTable = mTable.replace( QStringLiteral( " " ), QString() ).right( 63 );
+
+  mGeomColumn = parameterAsString( parameters, QStringLiteral( "GEOMETRY_COLUMN" ), context );
+  if ( mTable.isEmpty() )
+    mGeomColumn = QStringLiteral( "geom" );
+  if ( mSource->wkbType() == QgsWkbTypes::NoGeometry )
+    mGeomColumn = QString(); //port note: equivalent to initializing with python None
+
+  mCreateIndex = parameterAsBoolean( parameters, QStringLiteral( "CREATEINDEX" ), context );
+
+  if ( mOverwrite )
+    mOptions[QStringLiteral( "overwrite" )] = true;
+  if ( parameterAsBoolean( parameters, QStringLiteral( "LOWERCASE_NAMES" ), context ) )
+    mOptions[QStringLiteral( "lowercaseFieldNames" )] = true;
+  if ( parameterAsBoolean( parameters, QStringLiteral( "DROP_STRING_LENGTH" ), context ) )
+    mOptions[QStringLiteral( "dropStringConstraints" )] = true;
+  if ( parameterAsBoolean( parameters, QStringLiteral( "FORCE_SINGLEPART" ), context ) )
+    mOptions[QStringLiteral( "forceSinglePartGeometryType" )] = true;
+  if ( !mEncoding.isEmpty() )
+    mOptions[QStringLiteral( "fileEncoding" )] = mEncoding;
+
+  return true;
+}
+
+QVariantMap QgsExportToPostgresqlAlgorithm::processAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback )
+{
+  Q_UNUSED( parameters );
+  Q_UNUSED( context );
+
+  QgsDataSourceUri uri = QgsDataSourceUri( mConn->uri() );
+  uri.setSchema( mSchema );
+  uri.setTable( mTable );
+  uri.setKeyColumn( mPrimaryKeyField );
+  uri.setGeometryColumn( mGeomColumn );
+
+  mExporter.reset( new QgsVectorLayerExporter( uri.uri(), mProviderName, mSource->fields(), mSource->wkbType(), mSource->sourceCrs(), mOverwrite, mOptions ) );
```

Comment:
This doesn't need to be a member variable
```suggestion
  std::unique_ptr< QgsVectorLayerExporter > exporter = std::make_unique< QgsVectorLayerExporter >( uri.uri(), mProviderName, mSource->fields(), mSource->wkbType(), mSource->sourceCrs(), mOverwrite, mOptions );
```